frontend/packages/client/src/hooks/useNotificationService.js

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react';
+
+export default function useNotificationService() {
+  const [notificationSettings, setNotificationSettings] = useState({
+    walletId: '',
+    email: '',
+    communitySubscription: [],
+    isUnsubscribedFromCommunityUpdates: false,
+  });
+
+  const setUserID = async (walletId) => {
+    try {
+      //here we call api
+      setNotificationSettings((prevState) => ({
+        ...prevState,
+        walletId,
+      }));
+    } catch {
+      throw new Error('cannot set user id for leanplum');
+    }
+  };
+
+  const setUserEmail = async (email) => {
+    try {
+      //here we call api
+      setNotificationSettings((prevState) => ({
+        ...prevState,
+        email,
+      }));
+    } catch {
+      throw new Error('cannot set user email');
+    }
+  };
+  const getUserSettings = async () => {
+    try {
+      //here we call api
+      setNotificationSettings();
+    } catch {
+      throw new Error('cannot get user settings');
+    }
+  };
+
+  const unsubscribeCommunityNotifications = async (communityId) => {
```

Comment:
maybe rename to `unsubscribeCommunity` to have consistent function naming with `subscribeCommunity`